torch/optim/aggmo.py

```diff
@@ -0,0 +1,70 @@
+import torch
+from .optimizer import Optimizer, required
+
+
+class AggMo(Optimizer):
+    r"""Implements Aggregated Momentum Gradient Descent,
+    as proposed in `Aggregated Momentum: Stability Through Passive Damping`_.
+
+    Args:
+        params (iterable): iterable of parameters to optimize or dicts defining
+            parameter groups
+        lr (float): learning rate
+        momentum (list, optional): damping vector (default: [0, 0.9, 0.99])
+        weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
+
+    Example:
+        >>> optimizer = torch.optim.AggMo(model.parameters(), lr=0.1, momentum=[0,0.9,0.99,0.999])
+        >>> optimizer.zero_grad()
+        >>> loss_fn(model(input), target).backward()
+        >>> optimizer.step()
+
+    .. _Aggregated Momentum: Stability Through Passive Damping:
+        https://arxiv.org/abs/1804.00325
+    """
+
+    def __init__(self, params, lr=required, momentum=[0.0, 0.9, 0.99], weight_decay=0):
+        if not 0.0 <= lr:
+            raise ValueError("Invalid learning rate: {}".format(lr))
+        for i, mom in enumerate(momentum):
+            if not 0.0 <= momentum:
```

Comment:
This should be `mom`, not momentum. Also `mom` is a weird name for a variable. :p